reframe/core/environments.py

```diff
@@ -144,6 +144,23 @@ def emit_load_commands(*environs):
     return commands
 
 
+class modify_env:
```

Comment:
This is not how I meant to implement this context manager. You are not actually using the `Environment` class which has all the machinery we want. Here is how:

```python
class temp_environment:
    def __init__(modules=[], variables=[]):
        self._modules = modules
        self._variables = variables

    def __enter__(self):
        self._environ_save, _ =  load(Environment(self._modules, self._variables))
        return self._environ_save

    def __exit__(self, exc_type, exc_value, traceback):
        self._environ_save.restore()
```

And you should add a unit test for this.